libmultilabel/model.py

```diff
@@ -96,24 +120,25 @@ def test_epoch_end(self, step_outputs):
 
     def _shared_eval_step(self, batch, batch_idx):
         loss, pred_logits = self.shared_step(batch)
-        return {'loss': loss.item(),
-                'pred_scores': torch.sigmoid(pred_logits).detach().cpu().numpy(),
-                'target': batch['label'].detach().cpu().numpy()}
+        return {'loss': loss,
+                'pred_scores': torch.sigmoid(pred_logits),
+                'target': batch['label']}
 
     def _shared_eval_step_end(self, batch_parts):
-        pred_scores = np.vstack(batch_parts['pred_scores'])
-        target = np.vstack(batch_parts['target'])
-        return self.eval_metric.update(target, pred_scores)
+        return self.eval_metric.update(batch_parts['pred_scores'], batch_parts['target'])
 
     def _shared_eval_epoch_end(self, step_outputs, split):
-        metric_dict = self.eval_metric.get_metric_dict()
+        metric_dict = self.eval_metric.compute()
         self.log_dict(metric_dict)
+        for k, v in metric_dict.items():
+            metric_dict[k] = v.item()
         dump_log(metrics=metric_dict, split=split, log_path=self.log_path)
 
         if not self.silent and (not self.trainer or self.trainer.is_global_zero):
             print(f'====== {split} dataset evaluation result =======')
-            print(self.eval_metric)
-            print()
+            header = '|'.join([f'{k:^18}' for k in metric_dict.keys()])
+            values = '|'.join([f'{x * 100:^18.4f}' if isinstance(x, (np.floating, float)) else f'{x:^18}' for x in metric_dict.values()])
+            print(f"|{header}|\n|{'-----------------:|' * len(metric_dict)}\n|{values}|\n")
```

Comment:
Use ``self.print`` here?

Reply:
I removed `self.print()` previously because of several reasons:
1. We don't reuse `self.print()` in other places.
2. `self.print()` doesn't improve readability. We cannot tell the difference between `print()` and `self.print()` from the their function names.
3. As a `Model` class, having a `model.print()` function is uncommon and confusing to users. Some may consider `model.print()` will print out the model architecture.
Feel free to let me known if there is any consideration. Thanks.

Reply:
1. Indeed this is not a used elsewhere for now. But if a user want to print something in model, do we need to provide a short cut to do the errand ``if not self.silent and (not self.trainer or self.trainer.is_global_zero)``? 
2. & 3. For self.print() we follow PL's setting https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#print to print only from process 0 and additionally handle silent mode.  Indeed, the naming can be further discussed if that may be confusing.

Reply:
Sounds reasonable to me.
I think we can keep it `self.print()` to follow the convention.
I'll try to implement it by reusing the one provided in PL. (Our original version cannot take an empty string or multiple arguments.)